src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -34,6 +34,8 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
     -->
 
+  <Import Project="Microsoft.NET.$(PublishProtocol).targets" Condition="'$(PublishProtocol)' != '' and Exists('Microsoft.NET.$(PublishProtocol).targets')" />
```

Comment:
This cannot give a good error message if there is a typo in PublishProtocol. A space in it could be worse

Reply:
If there is a typo in the PublishProtocol, the ClickOnce targets file will not be found, the import will be skipped and ClickOnce publishing will not happen. That said, the PublishProtocol value is set by the Publish Provider in its schema file and not set by the user directly.

Reply:
> If there is a typo in the PublishProtocol, the ClickOnce targets file will not be found, the import will be skipped and ClickOnce publishing will not happen. 

This experience is hard to let the user trace back to the typo.

But if it is not set by the user that is fine. Do we have many PublishProtocol? and they are mutual exclusive?

Reply:
Yes there can be several publishing providers with their own unique protocol names and they are are mutually exclusive. The provider author specifies the protocol in a schema file (schema file is not directly available the to end user).